test/p1000e2e/boyar_driver.go

```diff
@@ -0,0 +1,116 @@
+package p1000e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Flaque/filet"
+	"github.com/orbs-network/boyarin/boyar/config"
+	"github.com/orbs-network/boyarin/services"
+	"github.com/orbs-network/boyarin/test/helpers"
+	"github.com/orbs-network/scribe/log"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+)
+
+const configJsonStr = `{
+  "network": [
+    {
+      "address":"dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173",
+      "ip":"127.0.0.1"
+    }
+  ],
+  "orchestrator": {
+    "max-reload-time-delay": "1s"
+  },
+  "chains": [
+    {
+      "Id":         {{.VChainId}},
+      "HttpPort":   8080,
+      "GossipPort": 4400,
+      "DockerConfig": {
+        "ContainerNamePrefix": "e2e",
+        "Image":  "orbs",
+        "Tag":    "export",
+        "Pull":   false
+      },
+      "Config": {
+        "active-consensus-algo": 2,
+        "genesis-validator-addresses" : [
+            "dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173"
+        ]
+      }
+    }
+  ],
+  "services": {}
+}`
+
+var configJsonTemplate = template.Must(template.New("").Parse(configJsonStr))
+
+type KeyConfig struct {
+	NodeAddress    string `json:"node-address"`
+	NodePrivateKey string `json:"node-private-key,omitempty"` // Very important to omit empty value to produce a valid config
+}
+
+func serveConfig(t *testing.T, vChainId int) *httptest.Server {
+	var sb strings.Builder
+	err := configJsonTemplate.Execute(&sb, struct {
+		VChainId int
+	}{
+		VChainId: vChainId,
+	})
+	require.NoError(t, err)
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		//logger.Info("configuration requested")
+		_, _ = fmt.Fprint(w, sb.String())
+	}))
+}
+
+/**
+set up environment and run boyar
+
+*/
+func InProcessBoyar(t *testing.T, logger log.Logger, keyPair KeyConfig, vChainId int) {
+	keyPairJson, err := json.Marshal(keyPair)
+	require.NoError(t, err)
+	file := filet.TmpFile(t, "", string(keyPairJson))
```

Comment:
Why do you have to introduce a dependency for creating a temporary file? Go already provides this functionality.